integration-tests/tests/src/tests/sets.ts

```diff
@@ -0,0 +1,493 @@
+          intSet: { type: "set", objectType: "int" },
+        },
+      };
+
+      const shorthandRealm = new Realm({ schema: [shorthandSchema] });
+      const shSchema = shorthandRealm.schema[0];
+      shorthandRealm.close();
+
+      const canonicalRealm = new Realm({ schema: [canonicalSchema] });
+      const canSchema = canonicalRealm.schema[0];
+      canonicalRealm.close();
+      expect(canSchema.properties).deep.equal(
+        shSchema.properties,
+        "Canonical and shorthand schemas should have idendical properties",
+      );
+    });
+  });
+
+  describe("General functionality", () => {
+    it("should add and delete operations", () => {
+      const realm = new Realm({ schema: [teamSchema] });
+
+      realm.write(() => {
+        // insert two people
+        realm.create(teamSchema.name, {
+          names: ["John", "Sue"],
+        });
+      });
+
+      let teams = realm.objects<Team>(teamSchema.name);
+      expect(teams.length).equals(1, "There should be one team");
+      expect(teams[0].names.size).equals(2, "Team Set size should be 2");
+
+      // add another three names
+      realm.write(() => {
+        teams[0].names.add("Pernille").add("Magrethe").add("Wilbur");
+      });
+      teams = realm.objects(teamSchema.name);
+      expect(teams.length).equals(1, "There should be one team.  We didn't add any");
+      expect(teams[0].names.size).equals(5, "Team Set size should be 5");
+
+      // remove two names
+      realm.write(() => {
+        teams[0].names.delete("John");
+        teams[0].names.delete("Sue");
+      });
+      teams = realm.objects(teamSchema.name);
+      expect(teams.length).equals(1, "There should still be only one team");
+      expect(teams[0].names.size).equals(3, "Team Set size should be 3");
+
+      realm.close();
+    });
+
+    it("should work with mixed types", () => {
+      const MixedObjectSchema = {
+        name: "MixedObject",
+        properties: {
+          mixedField: "mixed<>",
+        },
+      };
+
+      const IntSetObjectSchema = {
+        name: "IntObject",
+        properties: {
+          intField: "int",
+        },
+      };
+      interface MixedObject {
+        mixedField: Set<Realm.Types.Mixed>;
+      }
+      interface IntObject {
+        intField: Realm.Types.Int;
+      }
+
+      const realm = new Realm({ schema: [MixedObjectSchema, IntSetObjectSchema] });
+
+      const mixedValues: Realm.Types.Mixed[] = ["Joe", 54, true];
+      realm.write(() => {
+        const intObject = realm.create<IntObject>(IntSetObjectSchema.name, { intField: 41 });
+        mixedValues.push(intObject);
+      });
+
+      realm.write(() => {
+        realm.create(MixedObjectSchema.name, { mixedField: mixedValues });
+      });
+
+      const objs = realm.objects<MixedObject>(MixedObjectSchema.name);
+      expect(1).equals(objs.length, "One MixedObject");
+      expect(mixedValues.length).equals(objs[0].mixedField.size, `${mixedValues.length} values in set`);
```

Comment:
As with my previous comment, I'd suggest holding on to the created object instead of going through the `Results`.

```suggestion
      const obj = realm.write(() => {
        return realm.create(MixedObjectSchema.name, { mixedField: mixedValues });
      });

      expect(mixedValues.length).equals(obj.mixedField.size, `${mixedValues.length} values in set`);
```